README.md

```diff
@@ -24,58 +24,67 @@ pip3 install --upgrade datajoint
 ```
 ## Python Native Blobs
 
-For the v0.12 release, the variable `enable_python_native_blobs` can be
-safely enabled for improved blob support of python datatypes if the following
-are true:
+DataJoint 0.12 adds full support for all native python data types in blobs: tuples, lists, sets, dicts, strings, bytes, `None`, and all their recursive combinations.
+The new blobs are a superset of the old functionality and are fully backward compatible.
+In previous versions, only MATLAB-style numerical arrays were fully supported.
+Some Python datatypes such as dicts were coerced into numpy recarrays and then fetched as such.
 
-  * This is a new DataJoint installation / pipeline(s)
-  * You have not used DataJoint prior to v0.12 with your pipeline(s)
-  * You do not share blob data between Python and Matlab
+However, since some Python types were coerced into MATLAB types, old blobs and new blobs may now be fetched as different types of objects even if they were inserted the same way. 
+For example, new `dict` objects will be returned as `dict` while the same types of objects inserted with `datajoint 0.11` will be recarrays.
 
-Otherwise, please read the following carefully:
+Since this is a big change, we chose to disable full blob support by default as a temporary precaution, which will be removed in version 0.13.
+
+You can enable it by setting the `enable_python_native_blobs` flag in `dj.config`. 
+
+```python
+import datajoint as dj
+dj.config["enable_python_native_blobs"] = True
+```
+
+You can safely enable this setting if both of the following are true:
+
+  * All blobs in your current DataJoint databases contain only numerical arrays.
+  * You do not need to share blob data between Python and MATLAB.
```

Comment:
Why did the 'is a new pipeline'/'haven't used < 0.12' cases get removed?

Also: technically the 'squashed' data is 'a numerical array', and we haven't explained the details of squashing yet, so, devoid of details, making a statement based on an understanding of 'contain numerical arrays' is potentially dangerously ambiguous.

Put another way: the previous list outlined situations where the details of the disclaimer may need to be read, the new list outlines the cases where the data is correct, but we haven't described what those cases are cases yet..

Reply:
Having a new pipeline is not a necessary or sufficient requirement. It was not clear in the original formulation if all or any of the following conditions were required. I made the list so that it's **all** of the following conditions.

Reply:
This is not what I'm getting at - change can be fine, but it changes the intent of the block.

Previous intent: 

list the limited subset of cases for people who will never need to understand any details whatsoever, specifically intended to allow 'easy and not scary enable' strictly for new users who might be intimidated by the exception.  Everyone else should read the full details.

(So the 'insufficient' you mention was by design, to force the 'fall through' case of reading full and complete details for everyone but new-python-only users) 

Current intent (I surmise, please forgive): 

Summary of details for all users, if desired can read further

Are we confirming the change in intent?

If so, fine. 

However, if so, I don't think 'contain numerical arrays' is 'scary' enough to force reading the full details for those that should - Much like the phrase 'the following are true', 'contain numerical arrays' is  completely and precise in referring to a specific situation, but casually can be confusing without extra clarification. Since 'numpy array' is often used to casually refer to anything in the numpy package, and any of these things, including structured arrays are incorrectly used interchangably with the phrase 'numerical array' casually, I think we need be a bit more explicit here.

So, 2 things are needed for this discussion:

1) Confirm change in intent (apologies in harping, but this was not explicitly reflected in feedback)

2) Agree/Disagree on extending 'contain numerical array' text. If we are changing the intent, I would propose 'contain purely numerical arrays (see below for details)' to help provoke full reading by those who need to.

Reply:
we addressed this on Slack